slsim/lsst_science_pipeline.py

```diff
@@ -487,3 +498,274 @@ def cutout_image_psf_kernel(
+
+    :param radec: SpherePoint of radec
+    :type radec: an object: eg: geom.SpherePoint(65*degree, -36*degree)
+    :param dp0_image: image containing given radec
+    :return: corresponding Point2D of pixel coordinate in provided images.
+    """
+    if isinstance(dp0_image, list):
+        pix = []
+        for images in dp0_image:
+            pix.append(images.getWcs().skyToPixel(radec))
+    else:
+        pix = dp0_image.getWcs().skyToPixel(radec)
+    return pix
+
+
+def dp0_psf_kernels(pixel_coord, dp0_image):
+    """Extracts psf kernels of given dp0 image at given pixel coordinate.
+
+    :param pixel_coord: Pixel coordinate. eg:Point2D(2129.7674135086986,
+        2506.6640697199136)
+    :param dp0_image: dp0 image to extract psf
+    :return: list of psf kernels for given images
+    """
+    psf_kernels = []
+    for points, images in zip(pixel_coord, dp0_image):
+        dp0_image_psf = images.getPsf()
+        psf_kernels.append(dp0_image_psf.computeKernelImage(points).array)
+    return psf_kernels
+
+
+def dp0_time_series_images_data(butler, center_coord, radius="0.1", band="i", size=101):
+    """Creates time series data from dp0 data.
+
+    :param butler: butler object
+    :param center_coord: A coordinate point around which we need to create time series
+        images.
+    :param radius: radius for query
+    :param band: imaging band
+    :param size: cutout size of images
+    :return: An astropy table containg time series images and other information
+    """
+    expo_information = tap_query(center_coords=center_coord, radius=radius, band=band)
+    calexp_image = list_of_calexp(expo_information, butler=butler)
+    radec = dp0_center_radec(calexp_image[0])
+    radec_list = radec_list = [(radec.getRa().asDegrees(), radec.getDec().asDegrees())]
+    radec_list.extend(radec_list * (len(calexp_image) - 1))
+    cutout_image = calexp_cutout(calexp_image, radec, 450)
+    aligned_image = aligned_calexp(cutout_image)
+    aligned_image_cutout = calexp_cutout(aligned_image, radec, size)
+    pixels = radec_to_pix(radec, calexp_image)
+    psf_kernel = dp0_psf_kernels(pixels, calexp_image)
+    obs_time = expo_information["expMidptMJD"]
+    expo_time = expo_information["expTime"]
+    zero_point_mag = expo_information["zeroPoint"]
+    dp0_time_series_cutout = []
+    for i in range(len(aligned_image_cutout)):
+        dp0_time_series_cutout.append(aligned_image_cutout[i].image.array)
+    table_data = Table(
+        [
+            dp0_time_series_cutout,
+            psf_kernel,
+            obs_time,
+            expo_time,
+            zero_point_mag,
+            radec_list,
+        ],
+        names=(
+            "time_series_images",
+            "psf_kernel",
+            "obs_time",
+            "expo_time",
+            "zero_point",
+            "calexp_center",
+        ),
+    )
+    return table_data
+
+
+def variable_lens_injection(
```

Comment:
could you make this function being testable with an input?

Reply:
I added test functions for these functions.